src/common/utils/delete_documents.py

```diff
@@ -14,6 +14,25 @@ def delete_list_recursive(value: Union[list, dict], data_source: DataSource):
         delete_dict_recursive(value, data_source)
 
 
+def delete_by_attribute_path(document: dict, path: list[str], data_source: DataSource, get_data_source) -> dict:
+    target = document
+    path_elements = [e.strip("[]./") for e in path]
+    for element in path_elements[:-1]:  # Step through all the path items except the last one that should be deleted
+        if isinstance(target, list):
+            target = target[int(element)]
+        else:
+            target = target[element]
+
+    if isinstance(target, list):
+        delete_dict_recursive(target[int(path_elements[-1])], data_source=data_source)
+        del target[int(path_elements[-1])]
+    else:
+        delete_dict_recursive(target[path_elements[-1]], data_source=data_source)
```

Comment:
How do you know that `target[path_elements[-1]` is a dict? The function has no docstring stating that the path must lead to a dict and not a list

Reply:
Added as subissue in https://github.com/orgs/equinor/projects/189/views/3?reload=true&pane=issue&itemId=27214325